### PR TITLE
Remove behavior lookup

### LINKED ISSUE #3453: Remove behaviorLookup
### Description

 1. `Behaviors.behaviorLookup` is a left over of a time where was good practice to attach configuration to a specific namespace. The best practices today goes toward importing from modules as needed. In [docs]( https://marionettejs.com/docs/master/marionette.behaviors.html) its already deprecated.

Continuation of #3044

### PR DESCRIPTION
**Pull request by blikblum**
Fixes #3453 

Updated the tests to test defining behaviors both with array and object notation

### CODE REVIEW

**Comment by blikblum** [comment]
The https://github.com/marionettejs/backbone.marionette/pull/3462/commits/e5b6aab26236c848447c66606ebf43c2a5aa14c5 commit should have tested when option is invalid but coverage still marks this [part](https://coveralls.io/builds/13404906/source?filename=src%2Fmixins%2Fbehaviors.js#L22) as not executed

File: src/mixins/behaviors.js [hunk 1]
```diff
@@ -11,29 +11,24 @@ import Marionette from '../backbone.marionette';
...
-function getBehaviorClass(options, key) {
+// Otherwise an error is throw
+function getBehaviorClass(options) {
   if (options.behaviorClass) {
     return options.behaviorClass;
     //treat functions as a Behavior constructor
   } else if (_.isFunction(options)) {
     return options;
+  } else {
+    throw new MarionetteError('Unable to get behavior class. An Behavior constructor should be passed directly or as behaviorClass property of options');
```
Comment (paulfalgout) [inline src/mixins/behaviors.js]: nit pick  
```diff
+ A Behavior constructor
- An Behavior constructor
```

File: src/mixins/behaviors.js [hunk 2]
```diff
@@ -11,29 +11,24 @@ import Marionette from '../backbone.marionette';
 // If a user passes in options.behaviorClass
 // default to using that.
 // If a user passes in a Behavior Class directly, use that
-// Otherwise delegate the lookup to the users `behaviorsLookup` implementation.
-function getBehaviorClass(options, key) {
+// Otherwise an error is throw
```
Comment (JSteunou) [inline src/mixins/behaviors.js]: is throw or is thrown ? I would never know for sure

File: src/mixins/behaviors.js (hunk 1 above)
Comment (blikblum) [reply src/mixins/behaviors.js]: Changed

File: src/mixins/behaviors.js (hunk 2 above)
Comment (blikblum) [reply src/mixins/behaviors.js]: According to google translate is thrown. Changed

**Pull request merged**